lib/Sema/CSGen.cpp

```diff
@@ -1926,9 +1926,13 @@ namespace {
 
     /// Give each parameter in a ClosureExpr a fresh type variable if parameter
     /// types were not specified, and return the eventual function type.
-    Type getTypeForParameterList(ParameterList *params,
-                                 ConstraintLocatorBuilder locator) {
-      for (auto param : *params) {
+    Type getTypeForParameterList(ClosureExpr *closureExpr) {
+      auto *params = closureExpr->getParameters();
+      for (unsigned i = 0, n = params->size(); i != n; ++i) {
+        auto *param = params->get(i);
```

Comment:
Rather than this, I think you should be able to do:

```
  for (auto i : indices(params->getArray())) {
    auto *param = params->get(i);
```

Reply:
That's cool, will do!